### PR TITLE
Added Libxml, Librsvg, Gettext and Libadwaita demo

### PR DESCRIPTION
**Pull request by wroyca**
This PR add Libxml, Librsvg and Gettext which are needed to parse user interface (xml) files as well as to display icons and compile Libadwaita demo.

Screenshot_2022-10-30_142914

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
I would rather avoid to make installation instructions even more complex than they already are. Gettext will indeed be necessary as soon as internationalisation is introduced.

**Comment by wroyca** [comment]
> I would rather avoid to make installation instructions even more complex than they already are. Gettext will indeed be necessary as soon as internationalisation is introduced.

Granted, this makes the instruction a bit more complex, but I would argue that they are necessary, not optional. I doubt that anyone intends to use GTK and Libadwaita without a user interface. In fact, I was stung by this because `gtk_widget_class_set_template_from_resource` was failing to locate the resource — gresource was silently failing, and it looked like it was successfully generated. So yes, we can certainly compile a broken iteration of GTK4 and Libadwaita by skipping steps for simplicity, but I don't think the trade-off is worth it. 😅

**Comment by Hofer-Julian** [comment]
I tested every listing of the book on Windows, so I know that at least templates work.
However I've compiled GTK months ago, not sure if it introduced additional dependencies in the meantime.

Anyways, your PR is much appreciated @wroyca.
Let's see when I have time to reproduce it locally on my machine.
The test is whether I can compile a working libadwaita demo without your proposed changes, right?

**Comment by bilelmoussaoui** [comment]
> I can reproduce that icons don't appear without `librsvg`. Unfortunately, I get the following error when compiling it:
> 
> ```
>   = note: liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.14.rcgu.o) : error LNK2019: unresolved external symbol xmlInitParser referenced in function _ZN3std4sync4once4Once9call_once28_$u7b$$u7b$closure$u7d$$u7d$17h6d242d49ddf6dea3E.llvm.11956169904201746136
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.6.rcgu.o) : error LNK2019: unresolved external symbol xmlFreeDoc referenced in function _ZN4core3ptr56drop_in_place$LT$librsvg..xml..xml2_load..Xml2Parser$GT$17h399c368e8b7bed84E
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.10.rcgu.o) : error LNK2001: unresolved external symbol xmlFreeDoc
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.6.rcgu.o) : error LNK2019: unresolved external symbol xmlFreeParserCtxt referenced in function _ZN4core3ptr56drop_in_place$LT$librsvg..xml..xml2_load..Xml2Parser$GT$17h399c368e8b7bed84E
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.10.rcgu.o) : error LNK2001: unresolved external symbol xmlFreeParserCtxt
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.6.rcgu.o) : error LNK2019: unresolved external symbol xmlNewEntity referenced in function _ZN7librsvg3xml9xml2_load18sax_entity_decl_cb17h158f78582206afc2E
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.6.rcgu.o) : error LNK2019: unresolved external symbol xmlStopParser referenced in function _ZN7librsvg3xml9xml2_load23sax_start_element_ns_cb17hc9224348bb8b2563E
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.6.rcgu.o) : error LNK2019: unresolved external symbol xmlCreateIOParserCtxt referenced in function _ZN7librsvg3xml9xml2_load10Xml2Parser11from_stream17h83f835047b3875efE
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.6.rcgu.o) : error LNK2019: unresolved external symbol xmlCtxtUseOptions referenced in function _ZN7librsvg3xml9xml2_load10Xml2Parser11from_stream17h83f835047b3875efE
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.6.rcgu.o) : error LNK2019: unresolved external symbol xmlParseDocument referenced in function _ZN7librsvg3xml9xml2_load10Xml2Parser5parse17hd0b674667e9a4596E
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.6.rcgu.o) : error LNK2019: unresolved external symbol xmlCtxtGetLastError referenced in function _ZN7librsvg3xml9xml2_load10Xml2Parser5parse17hd0b674667e9a4596E
>           liblibrsvg-a52f19e459b0ac20.rlib(librsvg-a52f19e459b0ac20.librsvg.0980368a-cgu.10.rcgu.o) : error LNK2019: unresolved external symbol xmlFreeNode referenced in function _ZN7librsvg3xml8XmlState13entity_insert17h850452479ae74346E
>           C:\checkouts\personal-projects\librsvg\win32\vs16\release\x64\obj\rsvg_c_api\x86_64-pc-windows-msvc\release\deps\rsvg_convert.exe : fatal error LNK1120: 10 unresolved externals
> ```

looks like libxml wasn't installed properly

**Comment by Hofer-Julian** [comment]
> looks like libxml wasn't installed properly

Yeah, looks like it.
I've used the instructions in this PR, which look fine to me.

**Review by Hofer-Julian — APPROVED** [review]
Gettext was indeed the problem.
Two small comments, then it is ready to be merged from my side.
Thanks a lot @wroyca 

Seems like our windows instructions become more monstrous every time we touch it.
We should probably clarify somewhere that all of that is one big house of cards.

File: book/src/libadwaita.md [hunk 1]
```diff
@@ -59,6 +59,13 @@ From there, run the following commands:
...
 cd libadwaita
-meson setup builddir -Dprefix=C:/gnome -Dintrospection=disabled -Dvapi=false -Dexamples=false
+meson setup builddir -Dprefix=C:/gnome -Dintrospection=disabled -Dvapi=false
 meson install -C builddir
+
+# A workaround for https://gitlab.gnome.org/GNOME/gtk/-/issues/5303, which may
+# not be necessary in the future.
+#
+
+xcopy /s C:\gnome\share\icons\hicolor\scalable\apps C:\gnome\share\icons\hicolor\scalable\actions
```
[the remark this answers is not in the document]
Comment (wroyca) [reply book/src/libadwaita.md]: Done, thank!

**Comment by wroyca** [comment]
> Seems like our windows instructions become more monstrous every time we touch it.

Maybe we should set up a GitHub CI to provide a pre-built installation of the required GNOME libraries on Windows? I've been thinking about this for a while to automate GTK and Libadwaita installations 😓

**Comment by Hofer-Julian** [comment]
> > Seems like our windows instructions become more monstrous every time we touch it.
> 
> Maybe we should set up a GitHub CI to provide a pre-built installation of the required GNOME libraries on Windows? I've been thinking about this for a while to automate GTK and Libadwaita installations 😓

Would you be interested to set that up?
An installer like this would be quite helpful :)

**Comment by wroyca** [comment]
> > > Seems like our windows instructions become more monstrous every time we touch it.
> > 
> > 
> > Maybe we should set up a GitHub CI to provide a pre-built installation of the required GNOME libraries on Windows? I've been thinking about this for a while to automate GTK and Libadwaita installations 😓
> 
> Would you be interested to set that up? An installer like this would be quite helpful :)
Sure, I can look into it whenever I have time — I'll PR if I go through with it. (:

**Comment by Hofer-Julian** [comment]
Sounds good!

**Pull request merged**